redbot/setup.py

```diff
@@ -264,7 +264,8 @@ async def remove_instance(
 
     backend = get_current_backend(instance)
     driver_cls = drivers.get_driver_class(backend)
-    await driver_cls.initialize(**data_manager.storage_details())
+    if delete_data is True:
+        await driver_cls.initialize(**data_manager.storage_details())
```

Comment:
It would be better if all of this were under a single `if` statement along with `try/finally` clause:
```py
if delete_data is True:
    try:
        await driver_cls.initialize(**data_manager.storage_details())
        await driver_cls.delete_all_data(interactive=interactive, drop_db=drop_db)
    finally:
        await driver_cls.teardown()
```

Reply:
done